library/x509_crt.c

```diff
@@ -745,6 +777,83 @@ static int x509_get_subject_alt_name(unsigned char **p,
     return 0;
 }
 
+/*
+ * AuthorityKeyIdentifier ::= SEQUENCE {
+ *        keyIdentifier [0] KeyIdentifier OPTIONAL,
+ *        authorityCertIssuer [1] GeneralNames OPTIONAL,
+ *        authorityCertSerialNumber [2] CertificateSerialNumber OPTIONAL }
+ *
+ *    KeyIdentifier ::= OCTET STRING
```

Comment:
Missing information, check, and also a test for:
```
 -- authorityCertIssuer and authorityCertSerialNumber MUST both
    -- be present or both be absent
```

Reply:
Adding extra code to the parser to verify that both `authorityCertIssuer`, `authorityCertSerialNumber` are present or not doesn't matter too much. Also extra code needs to be added. For parser we should ensure that it won't crash if there is a case where for example `authorityCertIssuer` is present and `authorityCertSerialNumber` is not. From this perspective I think the current code is correct.

Function that generates or  verifies certificates should detect this case and return error.

Reply:
You're right!

Reply:
It’s possible to implicitly enforce this check: parse `authorityCertSerialNumber` as a mandatory field if `authorityCertIssuer` is present, and do not parse it at all if `authorityCertIssuer` is absent.